common/shell-local/run.go

```diff
@@ -58,31 +59,50 @@ func Run(ui packer.Ui, config *Config) (bool, error) {
 		if err != nil {
 			return false, err
 		}
-		ui.Say(fmt.Sprintf("Running local shell script: %s", script))
 
-		comm := &Communicator{
-			ExecuteCommand: interpolatedCmds,
+		// Check if command can execute against runtime.
+		runCommand := false
+		if len(config.OnlyOn) > 0 {
+			for _, os := range config.OnlyOn {
+				if os == runtime.GOOS {
+					runCommand = true
+				}
+			}
+		} else {
+			runCommand = true
 		}
 
-		// The remoteCmd generated here isn't actually run, but it allows us to
-		// use the same interafce for the shell-local communicator as we use for
-		// the other communicators; ultimately, this command is just used for
-		// buffers and for reading the final exit status.
-		flattenedCmd := strings.Join(interpolatedCmds, " ")
-		cmd := &packer.RemoteCmd{Command: flattenedCmd}
-		log.Printf("[INFO] (shell-local): starting local command: %s", flattenedCmd)
-		if err := cmd.StartWithUi(comm, ui); err != nil {
-			return false, fmt.Errorf(
-				"Error executing script: %s\n\n"+
-					"Please see output above for more information.",
-				script)
-		}
-		if cmd.ExitStatus != 0 {
-			return false, fmt.Errorf(
-				"Erroneous exit code %d while executing script: %s\n\n"+
-					"Please see output above for more information.",
-				cmd.ExitStatus,
-				script)
+		if runCommand {
```

Comment:
This works fine, but I think a more common pattern in this repo is to return early instead of wrapping the whole rest of the method in an if statement.  For example:
```
if !runCommand {
	ui.Say(fmt.Sprintf("Skipping local shell script due to runtime OS: %s", script))
	log.Printf("[INFO] (shell-local): skipping command due to runtime OS not specified.")
	return true, nil
}
ui.Say(fmt.Sprintf("Running local shell script: %s", script))
...
```

Reply:
Actually, yeah -- this should absolutely be at the very beginning before we even bother processing the inline stuff.

Reply:
it certainly doesn't need to be inside of the loop of scripts, since it'll be the same every time during this particular run.

Reply:
Moved it up to the entry point of func. I would say there is no point in doing any formatting/conversions if the command is only intended to run on specific OSs. Willing to move around though, if needed! Thanks again for the help/feedback! 👍